tests/run-make/env-dep-info/rmake.rs

```diff
@@ -0,0 +1,29 @@
+// Inside dep-info emit files, #71858 made it so all accessed environment
+// variables are usefully printed. This test checks that this feature works
+// as intended by checking if the environment variables used in compilation
+// appear in the output dep-info files.
+// See https://github.com/rust-lang/rust/issues/40364
+
+use run_make_support::{fs_wrapper, rustc};
+
+// FIXME(Oneirical): try on musl
+
+fn main() {
+    rustc()
+        .env("EXISTING_ENV", "1")
+        .env("EXISTING_OPT_ENV", "1")
+        .emit("dep-info")
+        .input("main.rs")
+        .run();
+    let dep_info = fs_wrapper::read_to_string("main.d");
+    assert!(&dep_info.contains("# env-dep:EXISTING_ENV=1"));
+    assert!(&dep_info.contains("# env-dep:EXISTING_OPT_ENV=1"));
+    assert!(&dep_info.contains("# env-dep:NONEXISTENT_OPT_ENV"));
```

Comment:
Question: this env var name confuses me. Does it mean `NONEXISTENT_OPT_ENV` does *not* exist and should *not* be contained in `--emit=dep-info`?

Reply:
No, it does exist, you can see it in the `main.rs` file. However, it is not assigned a value when `rustc` is called, so it's just printed with no value in the `.d` file.

Reply:
Ah I see, thanks